src/components/data/QueryAuth0.js

```diff
@@ -38,100 +40,117 @@ export default class QueryAuth0 extends React.PureComponent {
     userProfileReceived: PropTypes.func.isRequired,
   };
 
+  constructor(props) {
+    super(props);
+    this.state = {};
+  }
+
   async componentDidMount() {
-    const { authenticationFailed } = this.props;
+    this.checkForAuthentication();
+
+    window.addEventListener('focus', this.checkForAuthentication);
+  }
+
+  componentDidUpdate(prevProps) {
+    const { accessToken } = this.props;
+
+    if (accessToken && accessToken !== prevProps.accessToken) {
+      // Check the access token now
+      this.validateAccessToken();
+
+      // Set up periodic checks of the access token
+      this.setState(() => ({
+        valideAccessTokenInterval: window.setInterval(
```

Comment:
I don't think this needs to be inside a `this.setState(() => ...)` lambda, I think  just `this.setState({ key: value })` will work fine.

Also, I think  `valideAccessTokenInterval` is a typo, but I'm not sure for what.